astronomy/time_scales_test.cpp

```diff
@@ -360,6 +360,17 @@ TEST_F(TimeScalesTest, LunarEclipses) {
               AlmostEquals(91.2 * Second, 3e6, 4e6));
 }
 
+TEST_F(TimeScalesTest, JulianDate) {
+  EXPECT_THAT("JD2451545"_TT, Eq(j2000_week));
+  EXPECT_THAT("JD2455201.00000"_TT, Eq("2010-01-04T12:00:00.000"_TT));
+}
+
+TEST_F(TimeScalesTest, ModifiedJulianDate) {
+  EXPECT_THAT("MJD0.0"_TT, Eq("1858-11-17T00:00:00"_TT));
+  EXPECT_THAT("MJD55200.0000"_TT, Eq("2010-01-04T00:00:00.000"_TT));
+  EXPECT_THAT("MJD55200.00000142361"_TT, Eq("2010-01-04T00:00:00.123"_TT));
```

Comment:
In this implementation, nothing prevents MJD and JD from being used with UTC, [which poses problems](https://github.com/astropy/astropy/blob/master/cextern/erfa/ut1utc.c#L39-L42). I would not bother with supporting the ERFA quasi-JD convention, let's simply disallow those for UTC. That also makes it well-defined to go through `Time`.